tidy3d/components/medium.py

```diff
@@ -1267,6 +1336,257 @@ def elements(self) -> Dict[str, IsotropicMediumType]:
         return dict(xx=self.xx, yy=self.yy, zz=self.zz)
 
 
+class FullyAnisotropicMedium(AbstractMedium):
+    """Fully anisotropic medium.
+
+    Example
+    -------
+    >>> perm = [[2, 0, 0], [0, 1, 0], [0, 0, 3]]
+    >>> cond = [[0.1, 0, 0], [0, 0, 0], [0, 0, 0]]
+    >>> anisotropic_dielectric = FullyAnisotropicMedium(
+    >>>    permittivity=perm, conductivity=cond
+    >>>)
+    """
+
+    permittivity: RealTensor = pd.Field(
+        [[1, 0, 0], [0, 1, 0], [0, 0, 1]],
+        title="Permittivity",
+        description="Relative permittivity tensor.",
+        units=PERMITTIVITY,
+    )
+
+    conductivity: RealTensor = pd.Field(
+        [[0, 0, 0], [0, 0, 0], [0, 0, 0]],
+        title="Conductivity",
+        description="Electric conductivity tensor. Defined such that the imaginary part "
+        "of the complex permittivity at angular frequency omega is given by conductivity/omega.",
+        units=CONDUCTIVITY,
+    )
+
+    @pd.validator("permittivity", always=True)
+    def permittivity_spd_and_ge_one(cls, val):
+        """Check that provided permittivity tensor is symmetric positive definite
+        with eigenvalues >= 1.
+        """
+
+        if not np.allclose(val, np.transpose(val), atol=1.0e-10):
+            raise ValidationError("Provided permittivity tensor is not symmetric.")
+
+        if np.any(np.linalg.eigvals(val) < 1 - 1.0e-10):
```

Comment:
out of curiosity, what's the reasoning behind the 1e-10 addition? is it due to numerical error in the eigvals?

Reply:
yeah, exactly, sometimes values are just a little below 1